tasks/collector.py

```diff
@@ -36,6 +59,86 @@
 }
 
 
+class YAMLValidationError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def find_matching_components(manifest, components_to_match: dict, present: bool) -> list:
+    """Given a manifest and dict of components to match, if present=True, return list of
+    components found, otherwise return list of components missing."""
+    res = []
+    for component_type, components in components_to_match.items():
+        for component in components:
+            found_component = False
+            components_matching_component_type = manifest.get(component_type)
+            if components_matching_component_type:
+                for module in components_matching_component_type:
+                    if module.get("gomod").find(component) != -1:
+                        found_component = True
+                        if not present:
+                            found_component = True
+                        else:
+                            res.append(component)
+                        break
+            if not present and not found_component:
+                res.append(component)
+    return res
+
+
+def validate_manifest(manifest) -> list:  # return list of components to remove, empty list if valid
+    # validate collector version matches ocb version
+    manifest_version = manifest.get("dist").get("otelcol_version")
```

Comment:
```suggestion
    manifest_version = manifest.get("dist", {}).get("otelcol_version")
```
Or either
```
    manifest_version = manifest["dist"].get("otelcol_version")
```
If there are always `dist: otelcol_version:` you can even do that:
```
    manifest_version = manifest["dist"]["otelcol_version"]
```
Otherwise if `dist` is not present, a NoneType error will be raised

Reply:
you're correct, we are not necessarily guaranteed to have a dist section. I have made the change suggested.